### PR TITLE
[inductor]: improve support for pow codegen to reciprocal powers of 2

### PR DESCRIPTION
**Pull request by jon-chuang**
fixes spda autocast fused attention codegen on my local (i.e. `test_autocast_sdpa_dynamic_shapes`)

Local compiled without flash attention nor mem-efficient attention

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov @ColinPeppler @yanboliang

### CODE REVIEW

File: torch/_inductor/codegen/common.py [hunk 1]
```diff
@@ -285,10 +286,18 @@ def _print_Pow(self, expr):
...
-            return self._helper_sqrt(base)  # type: ignore[attr-defined]
-        elif exp == -0.5:
-            return "1/" + self._helper_sqrt(base)  # type: ignore[attr-defined]
+
+        if abs(exp) < 1.0 and exp != 0:
+            pow2 = math.log(1 / abs(exp), 2)
+            # call nested sqrt
+            if pow2.is_integer():
+                for _ in range(int(pow2)):
+                    base = self._helper_sqrt(base)  # type: ignore[attr-defined]
```
Comment (peterbell10) [inline torch/_inductor/codegen/common.py:295]: At this point would it not make more sense to just codegen a call to `pow`. Replacing pow with a loop of sqrt calls doesn't really make sense performance-wise.

Comment (jon-chuang) [reply torch/_inductor/codegen/common.py:295]: In the case I was fixing, it was only handling the additional -0.25 case i.e. 1/sqrt(sqrt(x)). Maybe we can just add a max nested depth? I'm not too clear why codegen pow is such a bad idea for Triton. But cpp has the same restriction.

Comment (peterbell10) [reply torch/_inductor/codegen/common.py:295]: Maybe @jansel can comment but I believe there was a principled objection to having floating point in indexing calculations generally.  There is no reason we couldn't have a `_helper_pow` that works the same as `_helper_sqrt`.

Comment (jon-chuang) [reply torch/_inductor/codegen/common.py:295]: These are the times I measured in CUDA:

```
sqrt_2_pow_op in 0.366592 ms.
sqrt_3_pow_op in 0.206848 ms.
reciprocal_sqrt_2_pow_op in 0.185344 ms.
reciprocal_sqrt_3_op in 0.18432 ms.

sqrt_2_op in 0.319488 ms.
sqrt_3_op in 0.423936 ms.
reciprocal_sqrt_2_op in 0.431104 ms.
reciprocal_sqrt_3_op in 0.531456 ms.

```

Looks like powf is faster when the exponent is negative smaller and or negative

So yes, using pow is probably better.

**Pull request closed without merging**